index.js

```diff
@@ -2,10 +2,16 @@
 var KEBAB_REGEX = /[A-Z\u00C0-\u00D6\u00D8-\u00DE]/g;
 var REVERSE_REGEX = /-[a-z\u00E0-\u00F6\u00F8-\u00FE]/g;
 
-module.exports = exports = function kebabCase(str) {
-	return str.replace(KEBAB_REGEX, function (match) {
-		return '-' + match.toLowerCase();
+module.exports = exports = function kebabCase(str, leadingDashFlag) {
+	let kebab = str.replace(KEBAB_REGEX, function (match) {
+		return '-' + match
 	});
+  	let firstChar = kebab.charAt(1);
+  	return (
+      leadingDashFlag === false &&
+      firstChar.toUpperCase() === firstChar
+      	? kebab.slice(1).toLowerCase() : kebab.toLowerCase()
+    );
```

Comment:
Here you are mixing tabs and spaces...
And you are doing unnecessary calls to `toLowerCase()` at the end.

What about simply writing the feature like this instead:

```js
module.exports = exports = function kebabCase(str, stripLeadingDash) {
  const kebab = str.replace(KEBAB_REGEX, function (match) {
     return '-' + match.toLowerCase();
   });

  return stripLeadingDash && kebab[0] === '-' ? kebab.slice(1) : kebab;
};
```

To me that is more readable and more clearly does what it says.

What do you think?